oak_runtime/src/node/http/server.rs

```diff
@@ -358,6 +362,21 @@ impl HttpRequestHandler {
     }
 }
 
+/// Check if the request contains a `TRANSFER_ENCODING` header, and reject the request in that case
+/// by returning an error.
+fn validate_request(req: &HttpRequest) -> anyhow::Result<()> {
+    if req
+        .headers
+        .as_ref()
+        .and_then(|map| map.headers.get(&TRANSFER_ENCODING.as_str().to_string()))
+        .is_some()
```

Comment:
Why is all this necessary as opposed to just

```rust
req.headers.get(TRANSFER_ENCODING).is_some()
```

?

Reply:
Done! Changed the signature of `validate_request` to take the original `http::Request` instead of our internal representation, in which `headers` is `Option<HeaderMap>`.